api/src/devcomp/domain/usecases/record-passage-events.js

```diff
@@ -0,0 +1,37 @@
+import { DomainError } from '../../../shared/domain/errors.js';
+import { CONCURRENCY_HEAVY_OPERATIONS } from '../../../shared/infrastructure/constants.js';
+import { PromiseUtils } from '../../../shared/infrastructure/utils/promise-utils.js';
+import {
+  FlashcardsCardAutoAssessedEvent,
+  FlashcardsRectoReviewedEvent,
+  FlashcardsRetriedEvent,
+  FlashcardsStartedEvent,
+  FlashcardsVersoSeenEvent,
+} from '../models/passage-events/flashcard-events.js';
+
+const recordPassageEvents = async function ({ events, passageEventRepository }) {
+  const passageEvents = events.map(_buildPassageEvent);
+
+  await PromiseUtils.map(passageEvents, (passageEvent) => passageEventRepository.record(passageEvent), {
+    concurrency: CONCURRENCY_HEAVY_OPERATIONS,
+  });
```

Comment:
question : Est-ce qu'on veut vraiment paralléliser ces insert ?

Est-ce qu'on est suffisamment confiant avec le `occurredAt` pour garantir l'ordre des évents ?

question encore plus ouverte : Que se passe-t-il en cas de perte d'événements ?

Reply:
C'est à dire la perte d'évènements ? 

Dans le cas où un occurredAt est invalide, on jette une erreur ce qui n'insère pas la liste des events en base.
Dans le cas d'une erreur lors de l'insertion, en effet il n'y a pas de transaction ou de gestion d'erreurs dessus. Après est-ce qu'il y a besoin de ça ?

Reply:
Discuté ensemble, utiliser `PromiseUtils.mapSeries` : 
```suggestion
  await PromiseUtils.mapSeries(passageEvents, (passageEvent) => passageEventRepository.record(passageEvent), {
    concurrency: CONCURRENCY_HEAVY_OPERATIONS,
  });
```